File: 205.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.length() != t.length()) return false;
        map<char, char> t2s;
        map<char, char> s2t;
        for(int i=0; i<s.size(); i++){
           if(s2t.find(s[i]) == s2t.end() && t2s.find(t[i])==t2s.end()){
               s2t[s[i]] = t[i];
               t2s[t[i]] = s[i];
           }else{
               if(s2t[s[i]] != t[i] || t2s[t[i]] != s[i]){
                   return false;
               }
           }
        }
        return true;
    }
};

File: golang/205/205.go
package main

func isIsomorphic(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	s2t := make(map[byte]byte)
	t2s := make(map[byte]byte)
	for i := 0; i < len(s); i++ {
		if _, ok := s2t[s[i]]; ok {
			if s2t[s[i]] != t[i] {
				return false
			}
		}

		if _, ok := t2s[t[i]]; ok {
			if t2s[t[i]] != s[i] {
				return false
			}
		}

		s2t[s[i]] = t[i]
		t2s[t[i]] = s[i]
	}
	return true
}
